src/providers/twitch/TwitchAccount.cpp

```diff
@@ -257,41 +255,50 @@ void TwitchAccount::loadEmotes()
                           });
                 emoteData->emoteSets.emplace_back(emoteSet);
             }
+            // Getting userstate emotes from Ivr
+            // Without delaying this, deadlock occurs
+            QTimer::singleShot(1, [this] {
+                this->loadUserstateEmotes();
+            });
```

Comment:
The singleShot is only necessary because both `loadUSerstateEmotes` and our current scope locks `this->emotes_`. You can 'simply' scope the access to emoteData on line 216 so that it's unlocked before `loadUserstateEmotes` is called. See https://gist.github.com/pajlada/5376a2d3ea3abbc904aeaf83bc46d6c0

Reply:
Fixed this in [`8f691c0` (#2857)](https://github.com/Chatterino/chatterino2/pull/2857/commits/8f691c094d593acb22609584d99aad20d98a8b9a)